app/celer-g4/DetectorConstruction.cc

```diff
@@ -133,19 +134,19 @@ G4VPhysicalVolume* DetectorConstruction::Construct()
     }
     else if (field_type == "uniform")
     {
-        auto field = GlobalSetup::Instance()->GetMagFieldZTesla();
+        auto field = GlobalSetup::Instance()->GetMagFieldTesla();
         if (norm(field) > 0)
         {
             CELER_LOG_LOCAL(info)
-                << "Using a uniform field (0, 0, " << field[2] << ") in tesla";
+                << "Using a uniform field (0, 0, " << field[2] << ") [tesla]";
             mag_field_ = std::make_shared<G4UniformMagField>(
                 convert_to_geant(field, CLHEP::tesla));
         }
 
         // Convert field units from tesla to native celeritas units
         for (real_type& v : field)
         {
-            v *= units::tesla;
+            v = native_value_from(units::FieldTesla{v});
```

Comment:
You could also do the same for the conversion in celer-sim/Runner.cc while you're at it.